kubehelpers/psm.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 gRPC authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubehelpers
+
+import (
+	"fmt"
+
+	grpcv1 "github.com/grpc/test-infra/api/v1"
+)
+
+// IsPSMTest checks if a given Loadtest is a PSM test. This check must be
+// used after validate the spec of the clients.
+func IsPSMTest(clients *[]grpcv1.Client) bool {
+	for _, c := range *clients {
+		if ContainerForName("xds-server", c.Run) != nil {
+			return true
+		}
+	}
+	return false
+}
+
+// IsProxiedTest is to check if the current test have sidecar container
```

Comment:
"checks if the current test has a sidecar container specified."

"This check must be performed after validating the client specs."